react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -265,23 +265,54 @@ - (void)dealloc {
     [self performSelectorOnMainThread:@selector(invalidate) withObject:nil waitUntilDone:YES];
 }
 
+typedef JSGlobalContextRef (^JSContextRefExtractor)();
+
+void _initializeOnJSThread(JSContextRefExtractor jsContextExtractor) {
+    // Make sure the previous JS thread is completely finished before continuing.
+    static __weak NSThread *s_currentJSThread;
+    while (s_currentJSThread && !s_currentJSThread.finished) {
+        [NSThread sleepForTimeInterval:0.1];
+    }
+    s_currentJSThread = [NSThread currentThread];
+
+    // Close all cached Realms from the previous JS thread.
+    realm::_impl::RealmCoordinator::clear_all_caches();
+
+    RJSInitializeInContext(jsContextExtractor());
+}
+
 - (void)setBridge:(RCTBridge *)bridge {
     _bridge = bridge;
 
     static __weak RealmReact *s_currentModule = nil;
     [s_currentModule invalidate];
     s_currentModule = self;
 
-    id<RCTJavaScriptExecutor> executor = [bridge valueForKey:@"javaScriptExecutor"];
-
-    if ([executor isKindOfClass:NSClassFromString(@"RCTWebSocketExecutor")]) {
+    if ([bridge executorClass] == objc_lookUpClass("RCTWebSocketExecutor")) {
 #if DEBUG
         [self startRPC];
 #else
         @throw [NSException exceptionWithName:@"Invalid Executor" reason:@"Chrome debug mode not supported in Release builds" userInfo:nil];
 #endif
-    }
-    else {
+    } else if ([bridge isKindOfClass:objc_lookUpClass("RCTCxxBridge")]) {
+        // probe for the new C++ bridge in React Native 0.45+
+
+        __weak __typeof__(self) weakSelf = self;
+        __weak __typeof__(bridge) weakBridge = bridge;
+        [bridge performSelector:@selector(executeBlockOnJavaScriptThread:) withObject:^{
```

Comment:
`executeBlockOnJavaScriptThread:` isn't declared publicly on `RCTBridge` - it's a method on `RCTCxxBridge` which is privately-declared.